modules/data-streams/src/internalClusterTest/java/org/elasticsearch/datastreams/logsdb/qa/StandardVersusLogsIndexModeChallengeIT.java

```diff
@@ -0,0 +1,347 @@
+    }
+
+    private static void mappings(final XContentBuilder builder) throws IOException {
+        builder.startObject("host.name")
+            .field("type", "keyword")
+            .field("ignore_above", 1024)
+            .endObject()
+            .startObject("message")
+            .field("type", "keyword")
+            .field("ignore_above", 1024)
+            .endObject()
+            .startObject("method")
+            .field("type", "keyword")
+            .field("ignore_above", 1024)
+            .endObject()
+            .startObject("memory_usage_bytes")
+            .field("type", "long")
+            .field("ignore_malformed", true)
+            .endObject();
+    }
+
+    @Override
+    public void oracleMappings(final XContentBuilder builder) throws IOException {
+        mappings(builder);
+    }
+
+    @Override
+    public void challengeMappings(XContentBuilder builder) throws IOException {
+        mappings(builder);
+    }
+
+    @Override
+    public void challengeSettings(Settings.Builder builder) {
+        builder.put("index.mode", "logs");
+    }
+
+    public void testMatchAllQuery() throws IOException, MatcherException {
+        final List<XContentBuilder> documents = new ArrayList<>();
+        int numberOfDocuments = randomIntBetween(100, 200);
+        for (int i = 0; i < numberOfDocuments; i++) {
+            documents.add(
+                XContentFactory.jsonBuilder()
+                    .startObject()
+                    .field(
+                        "@timestamp",
+                        DateFormatter.forPattern(FormatNames.STRICT_DATE_OPTIONAL_TIME.getName())
+                            .format(Instant.now().plus(i, ChronoUnit.SECONDS))
+                    )
+                    .field("host.name", randomFrom("foo", "bar", "baz"))
+                    .field("message", randomFrom("a message", "another message", "still another message", "one more message"))
+                    .field("method", randomFrom("put", "post", "get"))
+                    .field("memory_usage_bytes", randomLongBetween(1000, 2000))
+                    .endObject()
+            );
+        }
+
+        final Tuple<BulkResponse, BulkResponse> tuple = indexDocuments(() -> documents, () -> documents);
+        refresh();
+        assertThat(tuple.v1().hasFailures(), Matchers.equalTo(false));
+        assertThat(tuple.v2().hasFailures(), Matchers.equalTo(tuple.v1().hasFailures()));
+
+        final SearchSourceBuilder searchSourceBuilder = new SearchSourceBuilder().query(QueryBuilders.matchAllQuery())
+            .size(numberOfDocuments);
+        final SearchResponse oracleResponse = queryOracle(searchSourceBuilder);
+        final SearchResponse challengeResponse = queryChallenge(searchSourceBuilder);
+
+        new ResponseMatcher.Builder<>(getOracleMappings(), getOracleSettings(), getChallengeMappings(), getChallengeSettings()).with(
```

Comment:
mappings and settings are still not used here...but I think they are useful to print info in case of errors so that we can figure out differences which might result in test failures.